nikola/plugins/task/galleries.py

```diff
@@ -393,6 +411,55 @@ def create_galleries(self):
                 'uptodate': [utils.config_changed(self.kw.copy(), 'nikola.plugins.task.galleries:mkdir')],
             }
 
+    def find_metadata(self, gallery):
+        """Search for a gallery metadata file. Updates self if found."""
+        # If there is an metadata file for the gallery, use that to determine
+        # the order in which images shall be displayed in the gallery. The
+        # metadata file is YAML-formatted, with field names of
+        #
+        # name:
+        # caption:
+        # order:
+        #
+        # If order is specified, we use that directly, otherwise we depend on
+        # how PyYAML returns the information - which may or may not be in the
+        # same order as in the file itself. If no caption is specified, then we
+        # use an empty string.
+        # Returns a ordered list or None.
+
+        meta_path = os.path.join(gallery, "metadata.yml")
+        order = []
+        captions = {}
+        if os.path.isfile(meta_path):
+            self.logger.debug("Using {0} for gallery {1}".format(
+                meta_path, gallery))
+            self.meta_path = meta_path
+            with open(meta_path, "r") as meta_file:
+                meta = yaml.safe_load_all(meta_file)
+                for img in meta:
+                    # load_all and safe_load_all both return None as their
+                    # final element, so skip it
+                    if not img:
+                        continue
+                    if 'name' in img.keys():
```

Comment:
If that is a dict, you can use `if 'name' in img:` — and please do that in other `imgkeys` cases below

Reply:
Done.